components/director/internal/domain/formation/assign_formation_test.go

```diff
@@ -1330,6 +1397,23 @@ func TestServiceAssignFormation(t *testing.T) {
 			InputFormation:     inputFormation,
 			ExpectedErrMessage: testErr.Error(),
 		},
+		{
+			Name: "error when assigning tenant when formation does not support runtime",
+			FormationRepositoryFn: func() *automock.FormationRepository {
+				formationRepo := &automock.FormationRepository{}
+				formationRepo.On("GetByName", ctx, testFormationName, TntInternalID).Return(expectedFormation, nil).Once()
+				return formationRepo
+			},
+			FormationTemplateRepositoryFn: func() *automock.FormationTemplateRepository {
+				repo := &automock.FormationTemplateRepository{}
+				repo.On("Get", ctx, FormationTemplateID).Return(expectedFormationTemplateNotSupportingRuntime, nil)
+				return repo
+			},
+			ObjectType:         graphql.FormationObjectTypeTenant,
+			ObjectID:           TargetTenant,
+			InputFormation:     inputFormation,
+			ExpectedErrMessage: "Formation \"test-formation\" of type \"test-formation-template-name\" does not support resources of type \"TENANT\"",
```

Comment:
Use `fmt.Sprintf("...", parameters)` and reuse the variables for `formationName`, `formationTemplateName`, etc.. That way if someone decides to rename the variable, the test will work out of the box instead of manually changing the strings here